chalice/config.py

```diff
@@ -79,13 +80,15 @@ class Config(object):
 
     def __init__(self,
                  chalice_stage=DEFAULT_STAGE_NAME,
+                 function_name=DEFAULT_HANDLER_NAME,
                  user_provided_params=None,
                  config_from_disk=None,
                  default_params=None):
-        # type: (str, StrMap, StrMap, StrMap) -> None
+        # type: (str, str, StrMap, StrMap, StrMap) -> None
         #: Params that a user provided explicitly,
         #: typically via the command line.
         self.chalice_stage = chalice_stage
+        self.function_name = function_name
```

Comment:
I would actually see additional resources not using `Config` directly but instead using a resource specific config obj that `Config` can create.  Initially I had this as `Config --references-> LambdaConfig`, but reverted that in favor of this current approach because:

1) It's a bigger interface change across all the deployer code which is likely getting refactored soon anyways
2) Almost all the config in `Config` is actually lambda configuration.

So say if we had ddb tables, you'd say something like:

```
ddb_config = cfg.resource_config(cfg.stage_name, 'dynamodb_tables')
ddb_config['mytable']['name']
ddb_config['mytable']['provisioned_throughput']
... etc ...
```